ets-demo/etsdemo/loader.py

```diff
@@ -0,0 +1,121 @@
+# (C) Copyright 2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Module for loading example data files contributed by other distributions
+via entry points.
+"""
+
+import logging
+import os
+
+import pkg_resources
+
+from etsdemo.app import DemoPath, DemoVirtualDirectory
+
+
+logger = logging.getLogger(__name__)
+
+#: The entry point group name for other packages to contribute
+#: data files.
+ENTRY_POINT_GROUP = "etsdemo_data"
+
+
+def get_responses():
+    """ Load entry points and return the responses.
+
+    Version 1 response should have the following format:
+    {
+        "version": 1,
+        "name": <str>,
+        "root": <str>,
+    }
+
+    Required keys:
+    ``version``: int
+        The version for the format of this response
+    ``name``: str
+        The main display name for the group of resources.
+    ``root``: str
+        The root directory where resource files are stored.
+
+    Returns
+    -------
+    responses: list of dict
+        Responses as are loaded from the entry points.
+        Note that the content has not been sanitized nor validated.
+    """
+    request = {}
+    responses = []
+    for entry_point in pkg_resources.iter_entry_points(ENTRY_POINT_GROUP):
+        try:
+            loader = entry_point.load()
+        except ImportError:
+            logger.exception("Failed to load entry point %r", entry_point)
+            continue
+
+        try:
+            response = loader(request)
```

Comment:
The `except Exception` is already maximally defensive. Or maybe I misunderstood what you meant here.

Reply:
Sorry. I meant - do we want to define the signature of the `loader` right now and say that it should accept `*args, **kwargs` so that in the future, we can maintain backwards compatibility without needing updates in traitsui/pyface?

e.g. if the future, we pass a new argument to `loader` functions in `etsdemo` and we don't have to immediately update pyface/traitsui to accept the arg in their `loader` functions and make a new release.

Reply:
Ah, sorry I neglected to explain the intention of this `request` parameter.

It is currently an empty dict as a placeholder, for exact the reason of backward compatibility. The intention is that if in the future we need to supply more information to the `loader` function, instead of changing the function call signature, we will specify a schema for the request, like we specify a schema for the response.  In other words, the `loader` will always have just one argument.

Does this sound good?

Reply:
This makes me think I should add a comment here...